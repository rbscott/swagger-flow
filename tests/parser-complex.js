import yaml from 'js-yaml';

import parse from '../src/parser';

describe('parse', () => {
  const types = ['allOf', 'anyOf', 'oneOf'];

  describe('complexType', () => {

    types.forEach((type) => {
      it(`Should parse a ${type} complex type`, () => {
        const input = yaml.safeLoad(`
  definitions:
    complexType:
      ${type}:
      - type: object
        properties:
          field1:
            type: string
      - type: object
        properties:
          field2:
            type: string
  `);
        expect(parse(input).items).toMatchSnapshot();
      });
    });

    it('Should parse a few complex types with required fields', () => {
      const input = yaml.safeLoad(`
definitions:
  complexType:
    allOf:
    - type: object
      properties:
        field1:
          type: string
    - type: object
      properties:
        field2:
          type: string
    required:
      - field1
      - field2
`);
      expect(parse(input).items).toMatchSnapshot();
    });


  });
});
