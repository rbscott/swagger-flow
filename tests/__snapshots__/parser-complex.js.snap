// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parse complexType Should parse a allOf complex type 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field1",
            Object {
              "id": "#/definitions/complexType/SubType_0/Field_field1",
              "name": "complexType_SubType_0_Field_field1",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_0",
        "name": "complexType_SubType_0",
        "required": Array [],
        "type": "object",
      },
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field2",
            Object {
              "id": "#/definitions/complexType/SubType_1/Field_field2",
              "name": "complexType_SubType_1_Field_field2",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_1",
        "name": "complexType_SubType_1",
        "required": Array [],
        "type": "object",
      },
    ],
    "id": "#/definitions/complexType",
    "name": "complexType",
    "required": Array [],
    "type": "allOf",
  },
]
`;

exports[`parse complexType Should parse a anyOf complex type 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field1",
            Object {
              "id": "#/definitions/complexType/SubType_0/Field_field1",
              "name": "complexType_SubType_0_Field_field1",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_0",
        "name": "complexType_SubType_0",
        "required": Array [],
        "type": "object",
      },
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field2",
            Object {
              "id": "#/definitions/complexType/SubType_1/Field_field2",
              "name": "complexType_SubType_1_Field_field2",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_1",
        "name": "complexType_SubType_1",
        "required": Array [],
        "type": "object",
      },
    ],
    "id": "#/definitions/complexType",
    "name": "complexType",
    "required": Array [],
    "type": "anyOf",
  },
]
`;

exports[`parse complexType Should parse a few complex types with a ref 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field1",
            Object {
              "id": "#/definitions/complexType/SubType_0/Field_field1",
              "name": "complexType_SubType_0_Field_field1",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_0",
        "name": "complexType_SubType_0",
        "required": Array [],
        "type": "object",
      },
      Object {
        "id": "#/definitions/complexType/SubType_1",
        "name": "complexType_SubType_1",
        "ref": "#/definitions/otherType",
        "type": "ref",
      },
    ],
    "id": "#/definitions/complexType",
    "name": "complexType",
    "required": Array [],
    "type": "allOf",
  },
]
`;

exports[`parse complexType Should parse a few complex types with required fields 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field1",
            Object {
              "id": "#/definitions/complexType/SubType_0/Field_field1",
              "name": "complexType_SubType_0_Field_field1",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_0",
        "name": "complexType_SubType_0",
        "required": Array [],
        "type": "object",
      },
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field2",
            Object {
              "id": "#/definitions/complexType/SubType_1/Field_field2",
              "name": "complexType_SubType_1_Field_field2",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_1",
        "name": "complexType_SubType_1",
        "required": Array [],
        "type": "object",
      },
    ],
    "id": "#/definitions/complexType",
    "name": "complexType",
    "required": Array [
      "field1",
      "field2",
    ],
    "type": "allOf",
  },
]
`;

exports[`parse complexType Should parse a oneOf complex type 1`] = `
Array [
  Object {
    "choices": Array [
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field1",
            Object {
              "id": "#/definitions/complexType/SubType_0/Field_field1",
              "name": "complexType_SubType_0_Field_field1",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_0",
        "name": "complexType_SubType_0",
        "required": Array [],
        "type": "object",
      },
      Object {
        "additionalProperties": undefined,
        "fields": Array [
          Array [
            "field2",
            Object {
              "id": "#/definitions/complexType/SubType_1/Field_field2",
              "name": "complexType_SubType_1_Field_field2",
              "type": "string",
            },
          ],
        ],
        "id": "#/definitions/complexType/SubType_1",
        "name": "complexType_SubType_1",
        "required": Array [],
        "type": "object",
      },
    ],
    "id": "#/definitions/complexType",
    "name": "complexType",
    "required": Array [],
    "type": "oneOf",
  },
]
`;
